airflow/hooks/fs_hooks/_fnmatch.py

```diff
@@ -0,0 +1,137 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+
+"""Filename matching with shell patterns.
+
+Modified version of the stblib implementation that only matches '*.csv'
+in the current directory and supports recursive matching with '**/*.csv'.
+"""
+
+import os
+import posixpath
+import re
+import functools
+
+
+__all__ = ["filter", "fnmatch", "fnmatchcase", "translate"]
+
+
+def fnmatch(name, pat, sep=os.sep):
```

Comment:
How is this different to fnmatch from the stdlib?

Reply:
It allows for recursive globs, which the standard fnmatch does not allow. We can also not support recursive globbing, which allows for a simpler codebase.